File: android/app/src/main/kotlin/eagleteam/studentsocial/database/ScheduleDao.java
package eagleteam.studentsocial.database;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

import eagleteam.studentsocial.models.Schedule;
import eagleteam.studentsocial.models.User;

@Dao
public interface ScheduleDao {
    @Insert
    void insertListSchedules(List<Schedule> listSchedules);

    @Insert
    void insertOnlySchedule(Schedule schedule);

    @Update
    void updateOnlySchedule(Schedule schedule);

    @Delete
    void deleteOnlySchedule(Schedule schedule);

    @Query("select * from schedule_table where MaSinhVien = :msv")
    List<Schedule> getAllSchedules(String msv);

    @Query("delete from schedule_table")
    void deleteAllSchedules();

    @Query("select count(*) from schedule_table")
    int countSchedules();

    @Query("delete from schedule_table where MaSinhVien = :msv")
    void deleteScheduleByMSV(String msv);

    @Query("delete from schedule_table where MaSinhVien = :msv and LoaiLich != 'Note'")
    void deleteScheduleByMSVWithOutNote(String msv);

    @Query("select * from schedule_table where MaSinhVien = :msv and Ngay == :strDate")
    List<Schedule> getAllSchedulesFromDate(String msv, String strDate);
}

File: lib/models/local/database/schedule_dao.dart
import '../../../helpers/logging.dart';
import '../../entities/schedule.dart';
import 'database.dart';

class ScheduleDao {
  ScheduleDao(this.database);
  final MyDatabase database;

  Future<int> insertListSchedules(List<Schedule> listSchedules) async {
    logs('listSchedules is $listSchedules');
    for (int i = 0; i < listSchedules.length; i++) {
      await database.insert(listSchedules[i]);
    }
    //TODO: edit value return
    return 0;
  }

  Future<int> insertOnlySchedule(Schedule schedule) async {
    return await database.insert(schedule);
  }

  Future<int> updateOnlySchedule(Schedule schedule) async {
    return await database.updateSchedule(schedule);
  }

  Future<int> deleteOnlySchedule(Schedule schedule) async {
    return await database.deleteSchedule(schedule.MaSinhVien);
  }

  Future<List<Schedule>> getAllSchedule(String msv) async {
    return await database.getAllSchedule(msv);
  }

  Future<int> deleteAllSchedule() async {
    return await database.deleteAll(Schedule.table);
  }

  Future<void> countSchedules() async {
    await database.count(Schedule.table);
  }

  Future<int> deleteScheduleByMSV(String msv) async {
    return await database.deleteSchedule(msv);
  }

  Future<int> deleteScheduleByMSVWithoutNote(String msv) async {
    return await database.deleteScheduleWithoutNote(msv);
  }

  Future<List<Schedule>> getAllScheduleFromDate(String msv, String date) {
    return database.getAllScheduleFromDate(msv, date);
  }
}
